### PR TITLE
Use dedicated namespace instead of implicit (super)scope

### PR DESCRIPTION
**Pull request by timbru**
There is some duplication in this, but it's kind of hard to avoid that. I wanted a dedicated type for `Namespace` for type safety and clarity, but I also believe we should be even more restrictive in allowed characters here. There is also a macro. So, in short, I think it's not trivial to have a generic implementation with two flavours (e.g. using a marker phantom data).

After we merge this, we can add functionality to change the namespace of a `KeyValueStore` - or copy/backup, for that matter. Those things will be needed for upgrades and migrations in Krill.

### CODE REVIEW

File: kvx/src/implementations/postgres.rs [hunk 1]
```diff
@@ -85,57 +86,50 @@ impl<E: HasExecutor> KeyValueStoreBackend for Postgres<E> {
...
-                &[scope.as_vec(), &scope.len()],
+                "SELECT scope, key FROM store WHERE namespace = $1 AND scope[:$3] = $2",
+                &[&self.namespace, scope.as_vec(), &scope.len()],
             )?
             .into_iter()
             .map(|row| {
-                let scope: Vec<SegmentBuf> = row.get(0);
-                let mut scope = Scope::new(scope);
-                scope.remove_namespace(self.namespace.clone());
+                let scope_segments: Vec<SegmentBuf> = row.get(0);
```
Comment (partim) [inline kvx/src/implementations/postgres.rs:131]: I think you can merge these two lines into a simpl `Scope::new(row.get(0))`.

File: kvx/src/implementations/postgres.rs [hunk 2]
```diff
@@ -147,40 +141,35 @@ impl<E: HasExecutor> ReadStore for Postgres<E> {
...
             .into_iter()
             .flat_map(|row| {
-                let scope: Vec<SegmentBuf> = row.get(0);
-                let mut scope = Scope::new(scope);
-                if scope.remove_namespace(self.namespace.clone()).is_some() {
-                    scope.sub_scopes()
-                } else {
-                    vec![]
-                }
+                let scope_segments: Vec<SegmentBuf> = row.get(0);
```
Comment (partim) [inline kvx/src/implementations/postgres.rs:150]: Ditto ;)

**Review by partim — DISMISSED** [review]

File: kvx/src/implementations/memory.rs [hunk 3]
```diff
@@ -185,52 +152,42 @@ impl ReadStore for Memory {
...
         Ok(())
     }
 
     fn move_scope(&self, from: &Scope, to: &Scope) -> Result<()> {
-        let from = from.with_namespace(self.namespace.clone());
-        let to = to.with_namespace(self.namespace.clone());
-
         let mut inner = self.lock()?;
         *inner = inner
             .iter()
```
[the remark this answers is not in the document]
Comment (timbru) [reply kvx/src/implementations/memory.rs]: thanks, done!

File: kvx/src/implementations/postgres.rs (hunk 1 above)
Comment (timbru) [reply kvx/src/implementations/postgres.rs]: done

File: kvx/src/implementations/postgres.rs (hunk 2 above)
Comment (timbru) [reply kvx/src/implementations/postgres.rs]: done

**Pull request merged**